### PR TITLE
Hdrp/combine material samples with shader samples [Skip CI]

### PR DESCRIPTION
**Pull request by SeanPuller**
### Purpose of this PR

Merge material samples and shader samples so the similar assets are in a single place reducing unneeded sample folders. Align scene visual style and information with material samples to give users more information about how to use the materials/shaders provided.
Adding fabric materials and textures to give users many examples to cover all different types of fabric.

image

- Merged material samples and shader samples

- Update material samples scene visuals

- Added scenes for hair and fabric and decals with material samples

- Added fabric materials and textures

- Added information for fabric materials in fabric scene

- Fix Material Balls having same guid issue

- Fix spelling and grammatical errors in material samples

### CODE REVIEW

**Comment by iM0ve** [comment]
**Issues:**
* **Fixed**. Old package still visible
* **Fixed**. Weird looking displacement
* Files are poorly organized. This comes from before but I think we should fix this in this PR
  * We hav Meshes folder where only 1 object is an actual mesh
  * We have duplicate assets like MaterialBall.prefab in both Mesh and Prefab folders
  * We have assets that are not used like Nickel and Unlit materials. We should either implement them in the sample or delete the materials and its textures to reduce space
  * Overall folder structure looks messy, if possible simplify
* Since we use asset inside HDRP pipeline/RenderPipelineResources/Mesh/UnityBall.FBX.meta and we changed import settings. It is worth not skipping CI, since it could affect automated tests that have material ball in them
* The wording of the new explanations could be improved. Should consult with @JordanL8 (worth looking at all new text, below its just a sample). Simplify and clarify sentences wherever possible, avoid repetition.

image

* Optional. Maybe worth enabling micro shadows for the fabric sample? Im not sure how expensive this would be but imo gives a good bump in visuals for this scenario
JpnTsp1OyP

**Can be addressed in a separate PR (or here):**
* Hair sample does not leave a good impression in my opinion. Im not sure if its the shader or the presentation, but it feels off and unnatural. Maybe putting a wig on the ball and use shader on it would help? Not sure

image

File: com.unity.render-pipelines.high-definition/Documentation~/HDRP-Sample-Content.md [hunk 1]
```diff
@@ -18,14 +18,6 @@ Additional Post-Processing Data gives you access to Textures you can use with po
...
 
-## Shader Graph Samples
-
-This Sample includes example Shader Graphs that show you how to use the following Master Nodes: 
-
-* [Fabric Master Node](Master-Node-Fabric.html).
-* [Hair Master Node](Master-Node-Hair.html).
-* [Decal Master Node](Master-Node-Decal.html).
-
 The Fabric and Hair Master Nodes usually require various work from artists inside the Shader Graph and the Samples are a good head start.
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Documentation~/HDRP-Sample-Content.md:21]: This bit is from the Shader Graph Samples section. If you're removing that section content from there, I guess you want to remove this too?

File: com.unity.render-pipelines.high-definition/Documentation~/HDRP-Sample-Content.md [hunk 2]
```diff
@@ -40,5 +32,4 @@ This Sample includes various examples of lit and unlit particle effects.
 
 ![Material Samples](Images/MaterialSamples.png)
 
-This Sample includes various examples of Materials that use the [Lit Shader](Lit-Shader.html). The included Materials use effects such as subsurface scattering, displacement, and anisotropy. The **MaterialSamples** Scene requires Text Mesh Pro to display the text explanations.
-
+This Sample includes various examples of Materials. It includes materials that use the [Lit Shader](Lit-Shader.html), [Fabric Master Node](Master-Node-Fabric.html), [Hair Master Node](Master-Node-Hair.html) and [Decal Master Node](Master-Node-Decal.html). The included Materials use effects such as subsurface scattering, displacement, and anisotropy. The **MaterialSamples** Scene requires Text Mesh Pro to display the text explanations.
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Documentation~/HDRP-Sample-Content.md:35]: Just a couple of things:
* In the second sentence "materials" -> "Materials".
* Oxford comma missing after [Hair Master Node](Master-Node-Hair.html). `[Hair Master Node](Master-Node-Hair.html), and [Decal Master Node](Master-Node-Decal.html).`

File: com.unity.render-pipelines.high-definition/Documentation~/HDRP-Sample-Content.md (hunk 1 above)
Comment (SeanPuller) [reply com.unity.render-pipelines.high-definition/Documentation~/HDRP-Sample-Content.md]: moved it down under the material samples section

File: com.unity.render-pipelines.high-definition/Documentation~/HDRP-Sample-Content.md (hunk 2 above)
Comment (SeanPuller) [reply com.unity.render-pipelines.high-definition/Documentation~/HDRP-Sample-Content.md]: good catch

**Comment by JordanL8** [comment]
Here are my suggestions for the in-Scene text areas.

### Sections

**Cotton**
These Materials use the Fabric Shader with the Material Type set to **Cotton Wool**.

This is a diffused shading model which you can use to create Materials like cotton, denim, wool, linen, and velvet.

The type of fibers that make up the fabric, as well as the fabric's knit or weave, influence the appearance of the fabric. Natural fibers are typically rougher and therefore diffuse light.

**Silk**
These Materials use the Fabric Shader with the Material Type set to **Silk**.

You can use the **Silk** Material Type to create fabrics like silk, satin, nylon, and polyester.

In general, silk and synthetic fibers are smoother because they are produced as a single smooth filament. When this is weaved, it produces a fabric with anisotropic specular highlights.

### Individual Materials

**Cotton**
This Material uses a jersey knit threadmap typically used for clothing. It also includes a thin Diffusion Profile to add light transmission.

**Wool**
This Material uses a ribbed knit threadmap. This is a thicker knit with a higher ambient occlusion and normal strength.

**Denim**
This Material uses a fuzz map to simulate the color variation found in denim.

**Silk**
A smooth fabric with anisotropic specular highlights.

**Shot Silk**
Similar to **Silk** but with a separate specular color that gives it an iridescent look.

**Nylon**
Nylon is a synthetic material which is smoother than natural fabrics such as cotton or wool.

**Linen**
This Material uses a plain weave. It also uses a fuzz map to simulate color variation.

**Velvet**
Velvet can be made from synthetic or natural fibers. What gives velvet its unique appearance is its tufting. This means that velvet fibers point outwards from the surface which causes surfaces directly facing the viewer to appear darker.


### Threadmap

These Materials use a threadmap for the high-frequency details of fabrics. This is similar to the **Detail Map** found in HDRP's Lit Shaders.

The threadmap input is channel-packed to optimise memory and it arranged in a way to optimise precision for the normal map. Its format is:
* Red channel - Ambient occlusion
* Green and alpha channels - Normals
* Blue channel - Smoothness

**Comment by sebastienlagarde** [comment]
Triggered Yamato: https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/HDRP%252Fcombine_material_samples_with_shader_samples/.yamato%252Fupm-ci-abv.yml%2523all_project_ci_fast-2020.1/1938259/job/pipeline

**Comment by iM0ve** [comment]
As suspected, our fix for the material ball broke one automated test. We should update the reference image for the 1301_StackLitSG scene

**Comment by sebastienlagarde** [comment]
I will update it. I was expecting something like that, thus why I have trigger Yamato :)

**Pull request merged**